src/components/Calculator.js

```diff
@@ -20,36 +21,129 @@ Output.defaultProps = {
 };
 
 Output.propTypes = {
-  result: PropTypes.number,
+  result: PropTypes.string,
 };
 
-function Calculator() {
-  return (
-    <div className="calcContainer">
-      <Output result={0} />
-      <div className="keysGrid">
-        <Button className="calcBtn" value="AC" />
-        <Button className="calcBtn" value="+/-" />
-        <Button className="calcBtn" value="%" />
-        <Button className="calcBtn" value="÷" />
-        <Button className="calcBtn" value="7" />
-        <Button className="calcBtn" value="8" />
-        <Button className="calcBtn" value="9" />
-        <Button className="calcBtn" value="x" />
-        <Button className="calcBtn" value="4" />
-        <Button className="calcBtn" value="5" />
-        <Button className="calcBtn" value="6" />
-        <Button className="calcBtn" value="-" />
-        <Button className="calcBtn" value="1" />
-        <Button className="calcBtn" value="2" />
-        <Button className="calcBtn" value="3" />
-        <Button className="calcBtn" value="+" />
-        <Button className="calcBtn" value="0" />
-        <Button className="calcBtn" value="." />
-        <Button value="=" />
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  handleClick(value) {
+    const { total, next, operation } = this.state;
+    const calcObj = {
+      total,
+      next,
+      operation,
+    };
+    const newCalcObj = calculate(calcObj, value);
+    this.setState({
+      total: newCalcObj.total,
+      next: newCalcObj.next,
+      operation: newCalcObj.operation,
+    });
+  }
```

Comment:
- [ ] For this project, you're required to make use of the `useSatet` hook to manage the calculator app state. Kindly implement the use of useState hook. 👍🏽

HINT:
```javascript
// ./src/components/Calculator.js

import React, { useState } from 'react';
import calculate from ../logic; // Import math logic

function Calculator() {
    const [total, setTotal] = useState(0);
    return {
      <div>
       // Calculator app 
      </div>
    };
}
```

As required here:
<img width="1065" alt="Screenshot 2023-02-25 at 3 06 46 PM" src="https://user-images.githubusercontent.com/50941074/221364371-defd58dc-e028-4516-ac15-4997f2acd6e8.png">